### PR TITLE
feat(SmartAI): improve add quest action

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
##### CHANGES PROPOSED:

-  Rename ` SMART_ACTION_ADD_QUEST` to `SMART_ACTION_OFFER_QUEST`
-  This action can now open a dialog to offer the quest instead of adding it straight to the quest log
- With `action_param2` = `1` it will be like the old behaviour, with `0` it's the new behaviour
- All existing scripts will keep the same behaviour, except for creature `11216`

Feature by **@Treeston** imported from TC

###### ISSUES ADDRESSED:

- Updates https://github.com/azerothcore/azerothcore-wotlk/issues/1326

##### TESTS PERFORMED:

Builds

##### HOW TO TEST THE CHANGES:

- Test that the current scripts that use this action are not broken and use the same behaviour (with the exception of creature `11216` that will have the new behaviour). So check some existing scripts that add quests.
- Test the script of creature `11216` and make sure it asks for quest with a quest dialog

##### Target branch(es):

Master

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
sorry Treeston, I added the "@" before your name without actually meaning it. My only intention was to write your name to give credits (the notification was a mistake).

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -393,7 +393,7 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                                 menu.SendQuestGiverQuestDetails(q, me->GetGUID(), true);
 #if defined(ENABLE_EXTRAS) && defined(ENABLE_EXTRA_LOGS)
                                 sLog->outDebug(
-                                        "LOG_FILTER_DATABASE_AI",
+                                        LOG_FILTER_DATABASE_AI,
                                         "SmartScript::ProcessAction:: SMART_ACTION_OFFER_QUEST: Player guidLow %u - offering quest %u",
                                         (*itr)->GetGUIDLow(),
                                         e.action.questOffer.questID);
```
Comment (BarbzYHOOL) [inline src/server/game/AI/SmartScripts/SmartScript.cpp]: that looks horrible on github

Comment (FrancescoBorzi) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: what?

Comment (BarbzYHOOL) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: look

**Comment by wetbrownsauce** [comment]
It would seem that the only `smart_script` that uses `action_type` 7 is 11216, aka the one using the new behavior, so all testing I did had to involve that.  

To test, I first checked that this new behavior functions, then after that I changed it back to the old behavior and re-tested to make sure the old behavior still functions. Results as follows:

- New Behavior: When selecting the gossip option: "How can I help?" from Eva Sarkhoff (the option after going through her whole story), the gossip window closes and pops-up the quest dialog offering a quest. This all seems correct. 👍 
- Old Behavior: After changing `action_param2` back to 1, doing the above results in the quest automatically being added to the players quest log, with not pop-up quest offer dialog. So it seems the old behavior still functions as it did if `action_param2` is set to 1. 👍

**Comment by BarbzYHOOL** [comment]
@gtao725 would you like to help us test pull requests? If so, talk to me in the discord

**Pull request merged**